### PR TITLE
fix nil pointer in jobCache.update

### PR DESCRIPTION
**Pull request by Akiqqqqqqq**
Signed-off-by: Qiuyu Wu <qiuyu.wu@shopee.com>

fix: https://github.com/volcano-sh/volcano/issues/2417

### CODE REVIEW

File: pkg/controllers/cache/cache.go [hunk 1]
```diff
@@ -154,17 +154,19 @@ func (jc *jobCache) Update(obj *v1alpha1.Job) error {
 		return fmt.Errorf("failed to find job <%v>", key)
 	}
 
-	var oldResourceversion, newResourceversion uint64
-	var err error
-	if oldResourceversion, err = strconv.ParseUint(job.Job.ResourceVersion, 10, 64); err != nil {
-		return fmt.Errorf("failed to parase job <%v> resource version <%s>", key, job.Job.ResourceVersion)
-	}
+	if job.Job != nil {
```
Comment (hzxuzhonghu) [inline pkg/controllers/cache/cache.go:157]: When does job.Job=nil in update, i can see we set job.Job in Add callback

Comment (Akiqqqqqqq) [reply pkg/controllers/cache/cache.go:157]: as I mentioned in https://github.com/volcano-sh/volcano/issues/2417, when we delete vj, it triggers deleteJob() in informer of volcano-job-controller, deleteJob() calls cc.cache.Delete(job) which set **_jobInfo.Job = nil_** in **L182  pkg/controllers/cache/cache.go**

**Review by Thor-wl — APPROVED** [review]

**Comment by Thor-wl** [comment]
@Akiqqqqqqq Thanks for the fix. Can you also submit the fix to branch `release-1.6` `release-1.5` and `release-1.4`?

**Comment by Akiqqqqqqq** [comment]
> @Akiqqqqqqq Thanks for the fix. Can you also submit the fix to branch `release-1.6` `release-1.5` and `release-1.4`?

sure~

**Comment by Akiqqqqqqq** [comment]
> @Akiqqqqqqq Thanks for the fix. Can you also submit the fix to branch `release-1.6` `release-1.5` and `release-1.4`?

there is no such issue in 1.5 and 1.4, so I only submit the fix to 1.6

**Pull request merged**